.changeset/quiet-foxes-attend.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/react-i18n': minor
```

Comment:
> I'd consider the new behaviour to be something everyone should opt into without thinking about.

To me that pushes this change more towards being a `minor` rather than `major` release - major would be "you need to think about and plan for using this behaviour as it isn't trivial to drop in" which is sounds like this isn't :). The last variable is "does this require much test churn", and hopefully the answer is "not much".